src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -86,20 +82,23 @@ export const TradeConfirm = () => {
     slippage,
     buyAssetAccountId,
     sellAssetAccountId,
-    buyTradeAsset,
+    buyAmountCryptoPrecision,
   } = useFrozenTradeValues()
 
   const defaultFeeAsset = useAppSelector(state =>
     selectFeeAssetByChainId(state, trade?.sellAsset?.chainId ?? ''),
   )
 
-  const { bestSwapperWithQuoteMetadata } = useAvailableSwappers({ feeAsset: defaultFeeAsset })
-  const bestSwapper = bestSwapperWithQuoteMetadata?.swapper
+  const updateFiatSellAmount = useSwapperStore(state => state.updateSellAmountFiat)
+  const clearAmounts = useSwapperStore(state => state.clearAmounts)
+  const activeSwapperWithMetadata = useSwapperStore(state => state.activeSwapperWithMetadata)
+
+  const bestSwapper = activeSwapperWithMetadata?.swapper
 
   const reset = useCallback(() => {
-    swapperDispatch({ type: SwapperActionType.CLEAR_AMOUNTS })
-    swapperDispatch({ type: SwapperActionType.SET_VALUES, payload: { fiatSellAmount: '' } })
-  }, [swapperDispatch])
+    clearAmounts()
+    updateFiatSellAmount('')
```

Comment:
The atomic value setting seems a lot safer and definitely more ergonomic, nice one ser!